OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -182,20 +186,35 @@ void ConnectionStateChanged(OrderManager om, string password, NetworkConnection
 				{
 					var onSelect = new Action<string>(uid =>
 					{
-						// Don't select the same map again
-						if (uid == map.Uid)
+						// Don't select the same map again, and handle map becoming unavailable
+						if (uid == map.Uid && modData.MapCache[uid].Status != MapStatus.Available)
 							return;
 
 						orderManager.IssueOrder(Order.Command("map " + uid));
 						Game.Settings.Server.Map = uid;
 						Game.Settings.Save();
+						newMapUid = null;
+						oldMapUid = null;
+						lastUpdatedUid = null;
 					});
 
+					var onCancel = new Action(() =>
+					{
+						if (oldMapUid == map.Uid)
+							onSelect(newMapUid);
+						newMapUid = null;
+						oldMapUid = null;
+						lastUpdatedUid = null;
+					});
+
+					// We make sure lastUpdatedUid is up to date
+					modData.MapCache.UpdateMaps();
```

Comment:
If I'm following the logic here now, we're preselecting the map you've just changed on disk for your convenience? That seems handy.

I think a comment like `// Check for updated maps, if the user has edited a map we'll preselect it for them.` would be more descriptive as it's giving a 'why' instead of a 'what'.

Reply:
I'll change the comment right away